components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/fetch/GeckoViewFetchClient.kt

```diff
@@ -104,10 +104,11 @@ private fun WebRequest.Builder.addBodyFrom(request: Request): WebRequest.Builder
 }
 
 internal fun WebResponse.toResponse(isBlobUri: Boolean): Response {
+    val isDataUri = uri.startsWith("data:")
```

Comment:
Hm why are we checking `isDataUri` inside the method and `isBlobUri` on the [call site](https://github.com/mozilla-mobile/android-components/blob/master/components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt#L807)? Shouldn't we unify and do both at the same place?

I would probably move both checks to this `toReponse` extension method. Otherwise, the change looks good to me.

Reply:
I updated the pr :)